runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -178,7 +179,7 @@
   private transient PushedBackElementsHandler<WindowedValue<InputT>> pushedBackElementsHandler;
 
   // bundle control
-  private transient boolean bundleStarted = false;
+  private transient AtomicBoolean bundleStarted;
```

Comment:
Is there a scenario of multiple threads accessing this?

Reply:
Yes, there is a timer thread for the "finish bundle by time" feature, see `FlinkPipelineOptions#setMaxBundleTimeMills`.